### ISSUE TITLE
Cycle.\_\_getitem\_\_ has confusing semantics

### ISSUE THREAD

**Issue by jlstevens** [ioam/holoviews#983]
Current the implementation of ``Cycle.__getitem__`` is:

```python
 def __getitem__(self, num):
         return self(values=self.values[:num])
```

This acts as a slice even though the argument is an integer, not a slice. Integer indexing would be useful as that would make ``Cycle`` cyclic. This issue hasn't cropped up before as ``Cycle`` objects are rarely used directly.

I think this could be improved by supporting both integer indexing and slicing properly. This would be more flexible (using ``self.values[start:stop]``) and wouldn't require any new methods.

My concerns are that (1) this was implemented in 2c5bc0f in a way that looks like it is used by Palettes and I'm not sure about how this change would impact ``Palettes`` (2) issues regarding  backwards compatibility e.g if people used the existing style of slicing in notebooks.

**Comment by philippjfr** [comment]
This is bound up with another awkward bit of API on the ``Options`` object. The ``Options`` object defines a ``max_cycles`` method which truncates the Palette to the requested number of samples, which is necessary for Palettes to be discretely sampled correctly. If possible we should avoid ``Options`` from knowing abut ``max_cycles``, something probably best addressed at the same time as improving ``__getitem__`` semantics.

**Comment by jlstevens** [comment]
Is this an API change for holoviews 2.0? Or think we can tackle this before then?

**Comment by philippjfr** [comment]
I don't think this has ever been documented so I wouldn't be opposed to dealing with it before 2.0.
